app/models/alchemy/page.rb

```diff
@@ -114,10 +114,6 @@ class Page < ActiveRecord::Base
       if: -> { parent && parent.restricted? },
       unless: :systempage?
 
-    before_save :set_published_at,
-      if: -> { public? && published_at.nil? },
-      unless: :systempage?
```

Comment:
@tvdeyen I could not reuse this callback, as it never got called. `publish!` directly updates the columns, so `before_save` is not triggered. That's why the conditional had to go into the `publish!` method.